README.md

```diff
@@ -4,9 +4,54 @@ redirect_to:
 - LATEST
 ---
 
-# velero - Docs Branch
+# Velero - Docs Branch
 
 This special branch (`gh-pages`) is used to generate the [public documentation site](https://heptio.github.io/velero)
 for velero. This site is hosted using [Github Pages](https://help.github.com/articles/what-is-github-pages/).
 
 The content files in this branch were generated with a tool that Heptio builds and maintains. This means that if you change docs content in this branch to fix something on the docs site, make sure to change the equivalent content in the docs directory on the master branch.
+
+## View the site
+
+    In the root of the  project...
+
+    Run: `make serve`
+
+    Open http://0.0.0.0:4000/velero/
+
+## Test the site
+
+To test the site's docs for valid links and proper style, run the following in the root of the project:
+
+```
+make all-test
+```
+
+All test output files are written to the `./logs` directory (which is already in the `.gitignore`):
+* `link-check.log` - *[Checks the generated HTML]* Are links valid?
+
+* `style-check.log` - *[Checks the markdown source on `master`]* Are common documentation conventions followed?
+
+* `site-build.log` - Can the markdown source be properly converted into the HTML site? (The build output is used in the link checking test).
+
+Note: running `make clean` will delete the `logs` directory.
+
+## Where to make changes
+
+Where should changes go, given that there are files being pulled in from several different locations?
+
+* **Version-specific changes**
+  * *Ex:* fixing a typo or adding an image
+  * If changes need to be made inside a subdirectory like `v0.3.0/` or `master/`, they should be made as PRs to that tagged branch.
+  * **Do not** make version-specific changes to the `gh-pages` branch, since they may be overridden by the automated logic of `doc-site-gen`.
```

Comment:
This is no longer true; we should edit the docs in the `gh-pages` branch. This is because our gen_version.sh script no longer looks at a tag to generate docs.

If we kept the process as documented in this section, this would be the flow to get docs updated for specific versions:

* Checkout the release branch (such as `release-0.11`)
* Change the file in the `docs/` directory
* Push up a PR targeting the release branch
* Merge PR
* Cut a tag for the change, since `doc-site-gen` only works on tags
* Checkout the `gh-pages` branch
* Run `doc-site-gen update <tag>`
* Create PR for `gh-pages` branch.

If we just update docs on the `gh-pages` branch, the process is shorter:

* Checkout the `gh-pages` branch
* Make the change in the relevant directory/files
* Create PR for `gh-pages` branch.

I think the 2nd flow is less work, as well as allowing external contributors to contribute to the docs site without access to the `doc-site-gen` tool.

Reply:
Thanks for this take. I changed it to the second flow of course. Should I add that the same change needs to be made in the master branch under the `docs` dir for any change in the `master` folder?

Reply:
I think a force push may have overwritten a comment or it got deleted...but I think docs that apply to master should go there first, then get dumped from the master branch into the master folder; I'm mostly ok with the master copy on the site lagging a bit behind github.

I *think* it will be rare for us to update master and a specific version's docs before 1.0, but if we need that, then yes, we'll have to add to master and gh-pages branches.